enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/actions/AddADBAction.java

```diff
@@ -0,0 +1,144 @@
+                .getConnectedProfiles()
+                .stream()
+                .map(p -> p.getTenancy())
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+        Optional<TenancyItem> selectedTenancy = chooseOneItem(tenancies, Bundle.SelectTenancy());
+        
+        Optional<CompartmentItem> selectedCompartment = Optional.empty();
+                
+        if (selectedTenancy.isPresent()) {
+            List<CompartmentItem> compartments = CompartmentNode.getCompartments().apply(selectedTenancy.get());
+            selectedCompartment = chooseOneItem(compartments, Bundle.SelectCompartment());
+        }
+        Optional<DatabaseItem> selectedDatabase = Optional.empty();
+        
+        if (selectedCompartment.isPresent()) {
+            while(!selectedDatabase.isPresent()) {
+                OCIItem item = chooseCopartmentOrDb(selectedCompartment.get());
+                if (item == null) {
+                    return;
+                }
+                if (item instanceof DatabaseItem) {
+                    selectedDatabase = Optional.of((DatabaseItem) item);
+                }
+                if (item instanceof CompartmentItem) {
+                    selectedCompartment = Optional.of((CompartmentItem) item);
+                }
+            }
+        }
+        if (selectedDatabase.isPresent()) {
+            DownloadWalletAction action = new DownloadWalletAction(selectedDatabase.get());
+            action.actionPerformed(null);
+        }
+    }
+    
+    private <T extends OCIItem> Optional<T> chooseOneItem(List<T> ociItems, String title) {
+        Optional<T> result = Optional.empty();
+        if (ociItems.size() > 0) {
+            List<Item> items = ociItems.stream()
+                    .map(tenancy -> new Item(tenancy.getName(), tenancy.getDescription()))
+                    .collect(Collectors.toList());
+            NotifyDescriptor.QuickPick qp = new NotifyDescriptor.QuickPick(title, title, items, false);
+            if (DialogDescriptor.OK_OPTION == DialogDisplayer.getDefault().notify(qp)) {
+                Optional<String> selected = qp.getItems().stream().filter(item -> item.isSelected()).map(item -> item.getLabel()).findFirst();
+                if (selected.isPresent()) {
+                    result = ociItems.stream().filter(t -> t.getName().equals(selected.get())).findFirst();
+                }
+                
+            }
+        } else if (ociItems.size() == 1) {
+            result = Optional.of(ociItems.get(0));
+        }
+        return result;
+    }
+    
+    
+    private OCIItem chooseCopartmentOrDb(CompartmentItem compartment) {
+        List<OCIItem> items = new ArrayList<> ();
+        try {
+            items.addAll(DatabaseNode.getDatabases().apply(compartment));
+        } catch (BmcException e) {
+            LOGGER.log(Level.SEVERE, "Unable to load compartment list", e); // NOI18N
```

Comment:
Shouldn't be this reported somehow to the user ?

Reply:
This exception occurs in some cases when scanning the root compartment and doesn't affect the result